services/iota_wrappers.go

```diff
@@ -437,18 +437,51 @@ func verifyChunksMatchRecord(chunks []models.DataMap, checkTrunkAndBranch bool)
 	}
 
 	if response != nil && len(response.Hashes) > 0 {
-		trytesArray, err := api.GetTrytes(response.Hashes)
-		if err != nil {
-			oyster_utils.LogIfError(err, nil)
-			return filteredChunks, err
+
+		matchesTangle, notAttached, doesNotMatch := filterChunks(response.Hashes, chunks, checkTrunkAndBranch)
+
+		filteredChunks.MatchesTangle = append(filteredChunks.MatchesTangle, matchesTangle...)
+		filteredChunks.DoesNotMatchTangle = append(filteredChunks.DoesNotMatchTangle, doesNotMatch...)
+		filteredChunks.NotAttached = append(filteredChunks.NotAttached, notAttached...)
+
+	} else if len(response.Hashes) == 0 {
+		filteredChunks.NotAttached = chunks
+	}
+
+	if len(filteredChunks.MatchesTangle) > 0 {
+		oyster_utils.LogToSegment("iota_wrappers: chunks_matched_tangle", analytics.NewProperties().
+			Set("num_chunks", len(filteredChunks.MatchesTangle)))
+	}
+	if len(filteredChunks.NotAttached) > 0 {
+		oyster_utils.LogToSegment("iota_wrappers: not_attached", analytics.NewProperties().
+			Set("num_chunks", len(filteredChunks.NotAttached)))
+	}
+	return filteredChunks, err
+}
+
+func filterChunks(hashes []giota.Trytes, chunks []models.DataMap, checkTrunkAndBranch bool) (matchesTangle []models.DataMap,
+	notAttached []models.DataMap, doesNotMatch []models.DataMap) {
+
+	for i := 0; i < len(hashes); i += MaxNumberOfAddressPerFindTransactionRequest {
+		end := i + MaxNumberOfAddressPerFindTransactionRequest
+
+		if end > len(hashes) {
+			end = len(hashes)
 		}
 
-		transactionObjects := map[giota.Address][]giota.Transaction{}
+		if i >= end {
+			break
+		}
 
-		for _, txObject := range trytesArray.Trytes {
-			transactionObjects[txObject.Address] = append(transactionObjects[txObject.Address], txObject)
+		trytesArray, err := api.GetTrytes(hashes[i:end])
+
+		if err != nil {
+			oyster_utils.LogIfError(err, nil)
+			return matchesTangle, notAttached, doesNotMatch
```

Comment:
Was trying to avoid doing any further logic, if there wasn't anything in trytesArray.Trytes.  But I'll just add an explicit check for a length on trytesArray.Trytes.